### PR TITLE
Wait for payload sender to queue events in tests

### PR DESCRIPTION
**Pull request by russcam**
This PR 

1. Introduces wait methods to `MockPayloadSender` to allow waiting for events to be queued before making assertions
2. Removes `SerializerMockPayloadSender` which is no longer required now that sanitization happens in filters.

### CODE REVIEW

**Comment by russcam** [comment]
Jenkins please run the tests

**Review by gregkalapos — APPROVED** [review]
LGTM. Added some small things below.

File: test/Elastic.Apm.Tests/Mocks/MockPayloadSender.cs [hunk 1]
```diff
@@ -28,72 +28,164 @@ internal class MockPayloadSender : IPayloadSender
...
+		}
 
 		/// <summary>
-		/// The 1. Span on the 1. Transaction
+		/// Waits for transactions to be queued
 		/// </summary>
-		public Span FirstSpan => _spans.First() as Span;
+		/// <param name="timeout">Optional timeout to wait</param>
+		/// <returns><c>true</c> if the event was signalled, <c>false</c> otherwise.</returns>
+		public bool WaitForTransactions(TimeSpan? timeout = null, int? count = null)
```
[the remark this answers is not in the document]
Comment (russcam) [reply test/Elastic.Apm.Tests/Mocks/MockPayloadSender.cs]: Changed to be `void`. 

I've introduced some Signal methods that set the transaction or span `AutoResetEvent`. These can be called in cases where we're currently waiting for the absence of an event, where the order of execution is known, and after the invocation of code where we do not expect an event to be queued. Introducing these removes the need to arbitrarily wait a period of time and speeds up a number of tests.

**Comment by russcam** [comment]
run tests

**Comment by russcam** [comment]
Linux tests didn't run on the last commit and ended up timing out. Rerunning

**Comment by russcam** [comment]
run tests

**Comment by russcam** [comment]
There seems to be some an issue with CI test runs where certain stages are hanging; On the last run ,the `Test MsBuild` stage hung and on the run prior to that, the `Test Linux` stage hung. Since all the tests have passed in both stages across the two runs, I'm merging this in.

**Pull request merged**